.github/workflows/main.yml

```diff
@@ -14,19 +14,14 @@ jobs:
         fetch-depth: 0
         submodules: true
 
-    - uses: cachix/install-nix-action@v12
+    - uses: cachix/install-nix-action@v17
       with:
-        install_url: https://github.com/numtide/nix-flakes-installer/releases/download/nix-2.4pre20210823_af94b54/install
+        nix_path: nixpkgs=channel:nixos-unstable
```

Comment:
I'd consider using `nixos-22.05`, instead of `nixos-unstable`. Just to make CI a bit more predictable and performant. Same comment goes for `.github/workflows/pull.yml`. (I'm just nitpicking 😄).